scripts/sandbox.js

```diff
@@ -15,26 +18,51 @@ class Sandbox {
     });
     this.drawMarking(mouseX, mouseY);
   }
+
+  static redo() {
+    const changes = sandbox.changes;
+
+    if(changes.length - 1 > 0) {
+      changes.splice(changes.length - 1, 1);
+    }
+  }
+
   static pushLevel() {
+    const changes = sandbox.changes;
+
     scenes.splice(scenes.length - 1, 1, () => {
-      eval(sandboxLevel);
+      sandbox.level = changes[changes.length - 1];
+      eval(sandbox.level);
     });
   }
 
+  static updateTp(id) {
+    const inputElem = document.getElementById(id);
+    const content = inputElem.value;
+    console.log(id, sandbox.tpX);
+
+    if(id === "tpx") {
+      sandbox.tpX = eval(content);
```

Comment:
You can use computed properties here if you want. 😄 

```javascript
sandbox[id] = eval(content)
```

Reply:
Smart. I didn't think of that :p